### PR TITLE
Provide a generic database implementation.

### PR DESCRIPTION
**Pull request by mcastelino**
Make the networking database implementation generic so that it can be used to store data for any component in Ciao. This database implementation can be used to persist any table map, array, slice to a database backed by boltDB+gob.

We may want to consider moving this to a generic util package for ciao.

    **Note: As the table name internally has changed, this will break
    any in place update of the CN launcher. For a proper update
    of the CN, it has to be hard reset to remove all instances
    and the database cleaned up.**

### CODE REVIEW

**Comment by mcastelino** [comment]
@markdryan Could you review this please.

**Comment by markdryan** [comment]
@mcastelino , @obedmr We have almost identical database abstractions in this PR and in #632. We need to figure out which one to use, and also where to put it.

**Comment by obedmr** [comment]
@markdryan, as far I know we'd be using the one we're creating at #632, once it's done, a new PR would come for updating this networking code, there's only one missing piece that @mcastelino needs there in order to use it. @mcastelino could you confirm?

**Comment by mcastelino** [comment]
@obedmr It does not seem a completely transparent replacement right now. We can consider moving to your final implementation once you merge in. Also right now I need this to implement CNCI agent persistence.

**Comment by mcastelino** [comment]
@obedmr also in your implementation does the user of the database need to know that gob is used to store the value? I am going through your implementation, maybe I am mistaken, but I think the user of the db would need to decode?

File: networking/libsnnet/docker_database.go [hunk 1]
```diff
@@ -28,30 +28,62 @@ import (
...
+	Name() string
+
+	// Allocates and returns a single value in the table
+	NewElement() interface{}
+
+	// Add an value to the in memory table
+	Add(k string, v interface{}) error
+}
+
+// A TableDBProvider represents a persistent data base provider
```
Comment (mcastelino) [inline networking/libsnnet/docker_database.go]: @rbradford Good catch. fixed.

**Comment by mcastelino** [comment]
@rbradford fixed the review comments.

**Pull request merged**